### PR TITLE
[IJ Plugin] Suggest Apollo 4 migration from version catalog and build.gradle.kts dependencies

### PR DESCRIPTION
**Pull request by BoD**
Screenshot 2023-08-01 at 12 42 57

### CODE REVIEW

**Comment by martinbonnin** [comment]
What happens if the version is `3.8.1` above, do I still get the quick fix to bump to `3.8.2`? Or does the `migrate to Apollo 4` overrides that?

**Comment by BoD** [comment]
> What happens if the version is 3.8.1 above, do I still get the quick fix to bump to 3.8.2? Or does the migrate to Apollo 4 overrides that?

Both inspections and quickfixes are added on the same element:

---
Screenshot 2023-08-02 at 16 36 55

---
Screenshot 2023-08-02 at 16 37 11

(btw TDAIL (two days ago I learned) that the standard Gradle dependency inspection actually is powered by Android Lint)

File: intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/inspection/Apollo4AvailableInspection.kt [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            if (!version.startsWith("4")) {
+              holder.registerProblem(element, ApolloBundle.message("inspection.apollo4Available.reportText"), Apollo4AvailableQuickFix)
+            }
+          }
+        }
+      }
+
+      "implementation", "api", "testImplementation", "testApi" -> {
+        when (callExpression.valueArguments.size) {
+          // implementation("xxx:yyy:zzz")
```
[the remark this answers is not in the document]
Comment (BoD) [reply intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/inspection/Apollo4AvailableInspection.kt:97]: Actually this is already part of the v3->v4 migration :)

**Review by martinbonnin — APPROVED** [review]

**Comment by BoD** [comment]
(sorry I just made a change after you approved, to highlight the version reference in toml files rather than the dependency, to be consistent with the standard gradle dependency inspection)

**Review by martinbonnin — APPROVED** [review]

**Pull request merged**